driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -720,7 +719,7 @@ boolean checkSchemaAgreement() throws ConnectionException, BusyConnectionExcepti
         if (connection == null || connection.isClosed())
             return false;
 
-        DefaultResultSetFuture peersFuture = new DefaultResultSetFuture(null, cluster.protocolVersion(), new Requests.Query(SELECT_SCHEMA_PEERS));
+        DefaultResultSetFuture peersFuture = new DefaultResultSetFuture(null, cluster.protocolVersion(), new Requests.Query(SELECT_PEERS));
```

Comment:
`SELECT_PEERS` retrieves more columns, notably the tokens. So there's a tradeoff between the payload size (tokens can be non-negligible with virtual nodes), and considering bogus entries for schema agreement (which will probably make it fail). In the attachment to JAVA-852 there are lines that have a non-null rpc_address but other null fields, so using one query or the other would make a difference...